### PR TITLE
NEW make sell or eat by mandatory in product lot

### PR DESCRIPTION
**Pull request by lvessiller-opendsi**
NEW make sell or eat by mandatory in product lot (see PR #21709 for SQL install)

- Add a "make sell-by or eat-by date" mandatory in a product card when product batch is enabled
image

If sell-by or eat-by is mandatory then, we have these fields required when we create or update or product batch :
1) On create
image

2) On update
image

3) And an error message is shown if one of these fields are empty
image

4) We also check these mandatory fields on stock correction
image
- DLC and DLUO are automatically set when product lot is found

5) And check on dispatch from supplier order
image

6) And when reception is enabled we also have this check
image

### CODE REVIEW

**Comment by lvessiller-opendsi** [comment]
@eldy
I have to work on "checkSellOrEatByMandatoryFromProductIdAndDates()" and "checkSellOrEatByMandatoryFromProductIdAndDates" to put in centralized function create of stock movement.
Is it right ?

**Comment by eldy** [comment]
According to contributing rules, can you push the PR for database structure and migration into a independant PR ?

**Comment by lvessiller-opendsi** [comment]
@eldy 
> According to contributing rules, can you push the PR for database structure and migration into a independant PR ?

It had already been done in this PR : #21709

File: htdocs/product/ajax/product_lot.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+// Load Dolibarr environment
+require '../../main.inc.php'; // Load $user and permissions
+require_once DOL_DOCUMENT_ROOT.'/product/stock/class/productlot.class.php';
+
+$action = GETPOST('action', 'aZ09');
+$productId = GETPOST('product_id', 'int');
+$batch = GETPOST('batch', 'alphanohtml');
+
```
Comment (eldy) [inline htdocs/product/ajax/product_lot.php:46]: Missing test on permissions

**Comment by lvessiller-opendsi** [comment]
@eldy 
A pre-commit error still exists but I don't know why.
Could you check it ?

**Comment by lvessiller-opendsi** [comment]
@eldy
All checks have passed.

**Pull request merged**